### PR TITLE
Return a list of difficult words instead of just the count

### PR DESCRIPTION
**Pull request by rinaldo-rex**
The user can still get the total no. of difficult words, but enabling this makes sure it can be processed further in a user-friendly way. 

```python
>>>len(textstat.difficult_words(text_to_analyze)
```

🙂 Makes it more useful.

### CODE REVIEW

File: textstat/textstat.py [hunk 1]
```diff
@@ -231,9 +231,13 @@ def flesch_reading_ease(self, text):
         s_interval = 100 if self.__get_lang_root() in ['es', 'it'] else None
         syllables_per_word = self.avg_syllables_per_word(text, s_interval)
         flesch = (
-                206.835
-                - float(1.015 * sentence_length)
-                - float(84.6 * syllables_per_word)
+            self.__get_lang_cfg("fre_base")
```
Comment (alxwrd) [inline textstat/textstat.py]: Fixing merge conflict

**Comment by alxwrd** [comment]
@rinaldo-rex Thank you for this 🎉 

I agree 100% that this makes sense and would make the `difficult_words` method more useful. However, I'm worried about it being a breaking change.

For now, could you rename `difficult_words` to `difficult_words_list`, and keep your changes. Then add a new `difficult_words` that just returns `len(difficult_words_list(text))`?

```python
def difficult_words_list(self, text):
    # calculate difficult words
    return list()

def difficult_words(self, text):
    return len(self.difficult_words(text))
```

File: test.py [hunk 2]
```diff
@@ -228,7 +227,55 @@ def test_difficult_words():
     textstat.set_lang("en_US")
     result = textstat.difficult_words(long_test)
 
-    assert result == 49
+    assert set(result) == set(['exist',
```
Comment (alxwrd) [inline test.py]: Could we add a new test for `difficult_words_list()`, and use `short_test` instead?

```python
assert textstat.difficult_words_list(short_test) == ["sunglasses"] # I think?
```

File: textstat/textstat.py [hunk 3]
```diff
@@ -332,12 +330,12 @@ def difficult_words(self, text, syllable_threshold=2):
...
                 if self.syllable_count(value) >= syllable_threshold:
                     diff_words_set.add(value)
-        return len(diff_words_set)
+        return list(diff_words_set)
 
     @repoze.lru.lru_cache(maxsize=128)
     def dale_chall_readability_score(self, text):
         word_count = self.lexicon_count(text)
-        count = word_count - self.difficult_words(text)
+        count = word_count - len(self.difficult_words(text))
```
Comment (alxwrd) [inline textstat/textstat.py]: Don't forget to drop out these `len()`.

File: test.py (hunk 2 above)
Comment (rinaldo-rex) [reply test.py]: Done!

**Comment by rinaldo-rex** [comment]
I was actually thinking of making another method, but resorted back to changing the existing one! Hope this helps. I was actually using this lib to identify the difficult words and not just the count of them. So thought it would be a much-needed feature!

And great work with the lib btw! 🙂

**Comment by alxwrd** [comment]
Excellent. Thank you for this 🎉

**Pull request merged**